recipes/kwidgetsaddons/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class KwidgetsaddonsConan(ConanFile):
+    name = "kwidgetsaddons"
+    license = ("BSD-3-Clause", "LGPL-2.0-only", "LGPL-2.1-only", "LGPL-3.0-only", "LicenseRef-KDE-Accepted-LGPL", 
+        "GPL-2.0-or-later", "LGPL-2.0-or-later", "LGPL-2.1-or-later", "LGPL-3.0-or-later")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://api.kde.org/frameworks/kwidgetsaddons/html/index.html"
+    topics = ("conan", "desktop", "widgets", "qt")
+    description = "kwidgetsaddons contains add-on widgets and classes for applications that use the Qt Widgets module."
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def build_requirements(self):
```

Comment:
```suggestion
            del self.options.fPIC

    def configure(self):
        if self.options.shared:
            del self.options.fPIC

    def build_requirements(self):
```

Reply:
according to https://docs.conan.io/en/latest/mastering/conditional.html#constrain-settings-and-options `del self.options.fPIC` should be done in the `config_options` method

Reply:
Yes indeed, but there are 2 conditions on which `fPIC` is a *don't care*:
- on Windows: fPIC is not needed
- on Linux, when building a shared library. All code should be built as PIC, so shared=True and fPIC=False is an invalid configuration. Therefore, we remove fPIC when shared=True